src/rpc/core_rpc_server.cpp

```diff
@@ -273,6 +273,8 @@ namespace cryptonote
     m_net_server.set_threads_prefix("RPC");
     m_net_server.set_connection_filter(&m_p2p);
 
+    m_checkpoints.init_default_checkpoints(nettype());
+
```

Comment:
Can you not use the existing one in cryptonote_core (maybe a bit later if not initialized yet) ?

Reply:
Are you asking since it's a relatively expensive operation?

Reply:
Updated.

Reply:
I don't think it's that expensive. It just seems unnecessarily complex.